graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -538,18 +558,39 @@ def _rename_and_suppress_query_type_fields(
                         type named "Foo" will be unchanged iff type_renamings does not map "Foo" to
                         anything, i.e. "Foo" not in type_renamings
         query_type: name of the query type, e.g. 'RootSchemaQuery'
+        interfaces_to_make_unqueryable: interfaces to remove from the query type because one of
+                                        their descendants in the inheritance hierarchy was
+                                        suppressed.
 
     Returns:
         modified version of the input schema AST
 
     Raises:
         - SchemaTransformError if type_renamings suppressed every type
     """
-    visitor = RenameQueryTypeFieldsVisitor(type_renamings, query_type)
+    visitor = RenameQueryTypeFieldsVisitor(
+        type_renamings, query_type, interfaces_to_make_unqueryable
+    )
     renamed_schema_ast = visit(schema_ast, visitor)
     return renamed_schema_ast
 
 
+def _recursively_get_ancestor_interface_names(
```

Comment:
Necessary to define this function outside the `RenameSchemaTypesVisitor` object because we need access to the schema's `definitions` field in order to get the interface definition nodes.